File: MB-Chat/observability/safety-metrics.ts
import type { ObservabilityEvent } from "./types";

export function calculateSafetyMetrics(events: ObservabilityEvent[]) {
  return {
    safety_event_count: events.filter((event) => event.safety_flags.length > 0).length,
    phi_redaction_count: events.filter((event) => event.safety_flags.includes("phi_redacted")).length,
    unsafe_to_display_count: events.filter((event) => event.payload_summary.unsafe_to_display === true).length,
    blocked_output_count: events.filter((event) => event.event_type.includes("blocked")).length,
  };
}

File: MB-Secretaria/observability_py/safety_metrics.py
from .types import ObservabilityEvent


def calculate_safety_metrics(events: list[ObservabilityEvent]) -> dict[str, int]:
    return {
        "safety_event_count": len([event for event in events if event.safety_flags]),
        "phi_redaction_count": len([event for event in events if "phi_redacted" in event.safety_flags]),
        "unsafe_to_display_count": len([event for event in events if event.payload_summary.get("unsafe_to_display") is True]),
        "blocked_output_count": len([event for event in events if "blocked" in event.event_type]),
    }
